src/app/profiles.rs

```diff
@@ -22,7 +22,10 @@ pub struct Client<T> {
     context_token: String,
 }
 
-pub struct Rx(mpsc::Receiver<profiles::Routes>);
+pub struct Rx {
+    rx: mpsc::Receiver<profiles::Routes>,
+    _hangup: oneshot::Sender<()>,
```

Comment:
It would be useful to add an internal (`//` not `///`) comment explaining that it's just there to be dropped and we'll never actually send on it.

Reply:
If you follow @seanmonstar's `never` suggestion, the comment isn't really necessary as the type signature will be clearer about its use.